### PR TITLE
Add ability to put the assets in the flash

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

File: internal/backends/mcu/stm32h735g.rs [hunk 1]
```diff
@@ -99,6 +112,82 @@ pub fn init() {
...
+
+    let mut octospi =
+        dp.OCTOSPI1.octospi_unchecked(12.MHz(), &ccdr.clocks, ccdr.peripheral.OCTOSPI1);
+
+    octospi.configure_mode(OctospiMode::OneBit).unwrap();
+
+    // RDID Read Identification. Abuses address as instruction phase, but that
+    // works in SPI mode.
+    let mut read: [u8; 3] = [0; 3];
+    octospi.read(0x9F, &mut read).unwrap();
```
Comment (tronical) [inline internal/backends/mcu/stm32h735g.rs:138]: Is running this command necessary? I thought in the example this shows just how to run that command.

File: internal/backends/mcu/stm32h735g.rs [hunk 2]
```diff
@@ -99,6 +112,82 @@ pub fn init() {
...
+    octospi
+        .read_extended(
+            XW::U16(MX25LM51245G_OCTA_READ_CFG_REG2_CMD),
+            XW::U32(MX25LM51245G_CR2_REG1_ADDR),
+            XW::None,
+            5,
+            &mut read,
+        )
+        .unwrap();
+    assert_eq!(read[0], 1);
```
Comment (tronical) [inline internal/backends/mcu/stm32h735g.rs:163]: I’m curious, what is this checking?:)

File: internal/backends/mcu/stm32h735g/memory.x [hunk 3]
```diff
@@ -3,16 +3,27 @@
...
+       . = ALIGN(4);
        *(.frame_buffer);
        . = ALIGN(4);
      } > SDRAM
+     .slint_assets : {
+       . = ALIGN(4);
+      __s_slint_assets = .;
+       *(.slint_assets);
+       . = ALIGN(4);
+     } > SDRAM  AT>OSPI_ROM
```
Comment (tronical) [inline internal/backends/mcu/stm32h735g/memory.x:25]: Nice trick!

File: internal/backends/mcu/stm32h735g.rs [hunk 4]
```diff
@@ -90,7 +103,7 @@ pub fn init() {
 
     let hyperram_size = 16 * 1024 * 1024; // 16 MByte
     let config = hal::xspi::HyperbusConfig::new(80.MHz())
-        .device_size_bytes(24) // 16 Mbyte
+        .device_size_bytes(21) // 16 Mbyte
```
Comment (tronical) [inline internal/backends/mcu/stm32h735g.rs:106]: The value and the comment seem out of sync. Are you sure it’s only 2MB on your board?

I was under the impression that it’s 16MB and for example https://github.com/STMicroelectronics/STM32CubeH7/blob/c94252df7cec24a8fef67b28933353476a1edd3c/Projects/STM32H735G-DK/Examples/OSPI/OSPI_HyperRAM_MemoryMapped/Inc/main.h#L32 also uses 24.

**Review by tronical — APPROVED** [review]
Great solution! A few questions within :-)

File: internal/backends/mcu/stm32h735g.rs (hunk 4 above)
Comment (ogoffart) [reply internal/backends/mcu/stm32h735g.rs:106]: This was not intentional. Thanks for catching.

File: internal/backends/mcu/stm32h735g.rs (hunk 2 above)
Comment (ogoffart) [reply internal/backends/mcu/stm32h735g.rs:163]: that the write to the config register worked

File: internal/backends/mcu/stm32h735g.rs (hunk 1 above)
Comment (ogoffart) [reply internal/backends/mcu/stm32h735g.rs:138]: maybe it's not necessary, no.

**Pull request merged**